presto-main/src/main/java/com/facebook/presto/sql/planner/OrderingScheme.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.planner;
+
+import com.facebook.presto.spi.block.SortOrder;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+public class OrderingScheme
+{
+    private final List<Symbol> orderBy;
+    private final Map<Symbol, SortOrder> orderings;
+
+    @JsonCreator
+    public OrderingScheme(@JsonProperty("orderBy") List<Symbol> orderBy, @JsonProperty("orderings") Map<Symbol, SortOrder> orderings)
+    {
+        requireNonNull(orderBy, "orderBy is null");
+        requireNonNull(orderings, "orderings is null");
+        checkArgument(!orderBy.isEmpty(), "orderBy is empty");
+        checkArgument(orderings.keySet().equals(ImmutableSet.copyOf(orderBy)), "orderBy keys and orderings don't match");
+        this.orderBy = ImmutableList.copyOf(orderBy);
```

Comment:
Can `orderBy` have duplicates?

Reply:
I wouldn't assume it cant. Deduplication is `UnaliasSymbolReferences` job.

Reply:
AFAIRC there was a bug about that, because ANSI SQL allows that but Presto did not support that. I think @maciejgrzybek fixed that.